server/app/rest_api/routes/connector.py

```diff
@@ -0,0 +1,168 @@
+
+    @get(
+        "/content/{interface_id}/{resource_path:path}/{resource_name}",
+        operation_id="get_dataproduct_content",
+        name="Get Data Product Content",
+        tags=[Tags.Data_products],
+    )
+    async def get_dataproduct_content(
+        self, interface_id: str, resource_path: str, resource_name: str
+    ) -> StreamingResponse:
+        """Return the full dataset content (CSV)."""
+        return StreamingResponse(
+            iter([b"full,object,content\n1,2,3,4\n"] * 10),
+            media_type="text/csv",
+            headers={"Content-Disposition": f'attachment; filename="{resource_name}"'},
+            status_code=status.HTTP_200_OK,
+        )
+
+    @get(
+        "/content/{interface_id}/{resource_path:path}/{resource_name}/chunk",
+        operation_id="get_dataproduct_chunk",
+        name="Get Data Product Chunk",
+        tags=[Tags.Data_products],
+        responses={
+            200: {"description": "Full chunk when no range specified"},
+            206: {
+                "description": "Partial content chunk",
+                "content": {"text/csv": {"example": "partial,data\n"}},
+                "headers": {
+                    "Content-Range": {
+                        "description": "Range of bytes returned",
+                        "schema": {"type": "string", "example": "bytes=0-1023"},
+                    }
+                },
+            },
+            404: {"description": "Data product not found"},
+            416: {"description": "Range not satisfiable"},
+        },
+    )
+    async def get_dataproduct_chunk(
+        self,
+        interface_id: str,
+        resource_path: str,
+        resource_name: str,
+        start: Optional[int] = Query(None, description="Start byte position"),
+        end: Optional[int] = Query(None, description="End byte position"),
+    ) -> StreamingResponse:
+        """Return a dataset chunk (partial CSV content)."""
+        return StreamingResponse(
+            iter([b"partial,data\n"]),
+            media_type="text/csv",
+            headers={
+                "Content-Range": f"bytes={start}-{end}"
+                if start is not None and end is not None
+                else "bytes */*"
+            },
+            status_code=(
+                status.HTTP_206_PARTIAL_CONTENT
+                if start is not None and end is not None
+                else status.HTTP_200_OK
+            ),
+        )
+
+    @get(
+        "/metadata/dataproducts",
+        operation_id="list_dataproducts",
+        name="List Data Products",
+        tags=[Tags.Data_products],
+        response_model=List[schemas.DataProductSummary],
+    )
+    async def list_dataproducts(
+        self,
+        page: int = Query(1, ge=1, description="Page number"),
+        page_size: int = Query(10, ge=1, le=100, description="Items per page"),
+    ) -> JSONResponse:
+        """Return a paginated list of data products."""
+        all_data_products = [
```

Comment:
I think we should return the properties that is part of the ditrubtion, I mean for example we don't have ID in the distibution part of the ontology